orbit-2-savedstate/src/test/java/com/babylon/orbit2/SavedStatePluginWriteTest.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Babylon Partners Limited
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.babylon.orbit2
+
+import android.os.Parcel
+import android.os.Parcelable
+import androidx.lifecycle.SavedStateHandle
+import com.appmattus.kotlinfixture.kotlinFixture
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class SavedStatePluginWriteTest {
+
+    private val fixture = kotlinFixture()
+
+    @Test
+    fun `Modified state is saved in the saved state handle`() {
+        val initialState = fixture<TestState>()
+        val something = fixture<Int>()
+        val savedStateHandle = SavedStateHandle()
+        val middleware = Middleware(savedStateHandle, initialState)
+        val testStateObserver = middleware.container.orbit.test()
+
+        middleware.something(something)
+
+        testStateObserver.awaitCount(2)
+
+        assertThat(savedStateHandle.get<TestState?>(Container.SAVED_STATE_KEY)).isEqualTo(
+            TestState(something)
+        )
+    }
+
+    private data class TestState(val id: Int) : Parcelable {
```

Comment:
would be a bit overkill to depend on parcelize for this single test?

Reply:
Yeah which is why I did it the old way. Thankfully AS can generate this for you pretty easily